lib/messages.ex

```diff
@@ -0,0 +1,61 @@
+defmodule Messages do
```

Comment:
Why does your `Messages` module return tuples? What is the advantage of returning the atom `:menu` along with a given message? What if I want to use the message in a different place other than the menu? You would have to change the messages.

Reply:
By including the message type, Other modules can quickly apply effects or perform message processing based on the type of message it receives, if needed. In this case, it helps separate the Messages module from having to call colorizer function to apply color. Messages module has one and only one responsibility which is to return the message and it's type. Applying color is being handled by CLI display module.

Reply:
Messages are not bounced by the type. For example, messages can be extracted easily using pattern matching. Grouping messages is like putting functions with similar responsibilities in a module, but you can still call it from anywhere in your code.

Reply:
We talked about this a bit last week, but I will just reiterate what I mentioned here. By including the type of message with the message itself in a tuple, you are coupling a specific message to a specific type, making it difficult to reuse in different situations. For example, suppose you want to use these messages on a mobile app rather than in the console. You might use the same messages on different screens due to the navigation aspect of the mobile system. This means that your types would be inconsistent between the two interfaces.

Reply:
I understand that the type field might not be of much use in situations such as the one you mentioned above. The type is included with the message as extra information that can be ignored. Any program can invoke any functions in the Messages module without any limitation. If in a case of a mobile application where type is useless, it can be ignore and only the message be consumed. If however in case where for example, one wants to highlight errors in a web application and alert users, type can be very useful.

Reply:
I actually dislike including the type with the message for a different reason. IMO, the `type` is fairly coupled to the idea of a CLI. If you were to switch to a different UI, the `type` would very likely be at best ignored, and at worst misleading.

On the topic of reusing message, that actually goes against I18N best practices (which is effectively what you are doing by extracting the messages). The basic idea being that just because two places share the same text in one language, it does not necessarily follow that they will share the same text in every other language.

Another way of thinking about it: Extracting the messages is separating the content from the presentation of that content. The `type` is more related to the presentation than the content.

I recommend looking up I18N best practices.

Reply:
> If in a case of a mobile application where type is useless, it can be ignore and only the message be consumed.
Having part of a return value that can be ignored is often a code smell, and is not really that desirable.